kuma/core/context_processors.py

```diff
@@ -2,13 +2,35 @@
 
 from django.conf import settings
 from django.utils import translation
+from six.moves.urllib.parse import urlparse
 
 from .i18n import get_language_mapping
 
 
 def global_settings(request):
     """Adds settings to the context."""
-    return {'settings': settings}
+
+    def clean_safe_url(url):
+        if '://' not in url:
+            # E.g. 'elasticsearch:9200'
+            url = 'http://' + url
+        parsed = urlparse(url)
+        if '@' in parsed.netloc:
+            parsed = parsed._replace(
```

Comment:
I would prefer not using a "private" method on the parsed object. I was thinking that instead you could a standard string replacement on `parsed.netloc`, and then re-assemble the URL with `urlunparse`?

Reply:
It's been too long since I used Python 2 :)

The `_replace` isn't really a private API. It's [fully documented](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) **but only in Python 3's documentation!**
But it's **not** documented in [python 2's urlparse](https://docs.python.org/2/library/urlparse.html) so your point is correct.

However, if I leave it like this, this code will be ready for Python 3 without modification. With that in mind, would you be OK with leaving it like this so we can keep it when we go to Python 3?

Reply:
Wow, I didn't realize that `_replace` was documented! Thanks for pointing that out. It's really strange that they kept the `_` prefix in Python 3! Anyway, I'm totally OK with it knowing this, thanks!